lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -1009,19 +1010,16 @@ impl LocalShard {
         }
 
         // If still green while optimization conditions are triggered, mark as grey
-        if status == CollectionStatus::Green
+        if status == ShardStatus::Green
             && self.update_handler.lock().await.has_pending_optimizations()
+            && optimizer_status == OptimizersStatus::Ok
```

Comment:
Let's swap these because `has_pending_optimizations` can be a bit expensive and even takes a lock:

```suggestion
        if status == ShardStatus::Green
            && optimizer_status == OptimizersStatus::Ok
            && self.update_handler.lock().await.has_pending_optimizations()
```

Good suggestion by the way, to add the optimizer status check.